pkg/deploy/api/v1/conversion.go

```diff
@@ -196,6 +198,60 @@ func convert_api_DeploymentCauseImageTrigger_To_v1_DeploymentCauseImageTrigger(i
 	return nil
 }
 
+func convert_v1_RollingDeploymentStrategyParams_To_api_RollingDeploymentStrategyParams(in *RollingDeploymentStrategyParams, out *newer.RollingDeploymentStrategyParams, s conversion.Scope) error {
+	out.UpdatePeriodSeconds = in.UpdatePeriodSeconds
+	out.IntervalSeconds = in.IntervalSeconds
+	out.TimeoutSeconds = in.TimeoutSeconds
+	out.UpdatePercent = in.UpdatePercent
+
+	if in.UpdatePercent != nil {
+		pct := kutil.NewIntOrStringFromString(fmt.Sprintf("%d%%", *in.UpdatePercent))
+		if *in.UpdatePercent > 0 {
+			out.MaxSurge = pct
+		} else {
+			out.MaxUnavailable = pct
+		}
+	} else {
+		if err := s.Convert(in.MaxUnavailable, &out.MaxUnavailable, 0); err != nil {
+			return err
+		}
+		if err := s.Convert(in.MaxSurge, &out.MaxSurge, 0); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func convert_api_RollingDeploymentStrategyParams_To_v1_RollingDeploymentStrategyParams(in *newer.RollingDeploymentStrategyParams, out *RollingDeploymentStrategyParams, s conversion.Scope) error {
```

Comment:
Perhaps a summary here will help me.  Old clients asking for dcs with `MaxSurge` set, will try to round trip by dropping fields they don't recognize.  I'm about 90% sure that this means that a round trip through an old client will drop the `MaxSurge` and `MaxUnavailable` in a non-retrievable way and we aren't trying to save part of the information through `UpdatePercent`.

I don't have a problem with it just want to be sure.